### PR TITLE
feat: Order Book panel — free Coinbase L2 depth (crypto) + Alpaca L1 (stocks)

### PR DESCRIPTION
**Pull request by saikrishna-dev**
## Order Book / Market Depth panel

Adds a live order-book panel to the dashboard. Genuine free **Level-2** stock depth doesn't exist on any free tier (Polygon/Databento/Finnhub/Alpaca-SIP are all paid), so the data source is chosen **per symbol**:

| Symbol type | Source | Depth | Key needed? |
|-------------|--------|-------|-------------|
| Crypto pair (`BTC-USD`, `ETH-USD`…) | **Coinbase** free public API (`/products/{id}/book?level=2`) | **Real multi-level L2 ladder** | ❌ none — works out of the box |
| Stock (`NVDA`…) | Alpaca free IEX feed | Top-of-book (L1) | ✅ free Alpaca key (optional) |

Both paths emit the **same response shape** (multi-level `bids`/`asks` arrays + `spread`, `spread_pct`, `bid_ask_imbalance`, `mid_price`, `source`), so the frontend renders either without branching.

### Backend (`routers/market.py`)
- `GET /orderbook/{symbol}` routes by `_is_crypto_symbol()` (a `<BASE>-<QUOTE>` pair with a known quote currency → crypto).
- `_fetch_coinbase_book` / `_build_coinbase_book` — parses Coinbase's `[[price, size, num_orders], …]`, caps at 12 levels/side, computes a **depth imbalance** (aggregate resting size across levels). No API key.
- `_fetch_alpaca_book` / `_build_orderbook` — unchanged stock top-of-book; now returns 503 only for stocks when no Alpaca key, with a hint that crypto pairs work free.
- Redis-cached, 10s TTL.

### Frontend
- `OrderBookPanel.tsx` — depth ladder (`Bid Size | Price | Ask Size`), proportional green/red bars, mid-price, imbalance meter, 3s polling. Adds a small **source badge** (`Coinbase (L2)` / `Alpaca · IEX (L1)`).
- `app/api/orderbook/[symbol]/route.ts` proxy; rendered in the right sidebar of `page.tsx`.

### Env
- `backend/.env.example` — Alpaca is now **optional** (stocks only); documents Key-ID+Secret (not OAuth) and that crypto needs nothing.

### Tests & validation
- `tests/test_orderbook.py` — 10 tests: crypto-without-key → Coinbase success, `_is_crypto_symbol` routing, depth imbalance, depth cap, empty-book 404, plus the original Alpaca L1 cases.
- ✅ `pytest backend/tests/` — **69 passed**.
- ✅ `ruff check backend/` — clean.
- ✅ `pnpm run build` — TypeScript clean, exit 0.
- ⚠️ Live Coinbase call not exercised here (sandbox blocks outbound egress); parser matches Coinbase's documented payload shape — CI/preview have real egress.

https://claude.ai/code/session_01Ct4hheYmazzpaSuDxVvWhS

## Summary by CodeRabbit

* **New Features**
  * Level‑2 Order Book panel showing live multi‑level bids/asks, mid‑price, spread, and imbalance; auto-refreshes every 3s
  * Public orderbook API endpoint serving unified snapshots for crypto and equities, with short caching and backend proxy route

* **Documentation**
  * Added example backend environment variables and guidance for optional market/data and paper‑trading credentials; default PORT set

* **Bug Fixes / Reliability**
  * Improved provider error mapping and graceful responses for missing data or missing credentials

* **Tests**
  * New tests covering order book endpoint, data shaping, crypto routing, and edge cases

* **Chores**
  * Deployment workflows updated to pass new market API credentials to deployed services

### CODE REVIEW

File: .github/actions/setup-secrets/action.yml [hunk 1]
```diff
@@ -70,6 +70,18 @@ runs:
...
+      uses: koyeb/action-git-deploy/secret@v1
+      with:
+        secret-name: ALPACA_API_KEY
+        secret-value: "${{ env.ALPACA_API_KEY }}"
+
+    - name: Create ALPACA_SECRET_KEY secret
+      uses: koyeb/action-git-deploy/secret@v1
+      with:
+        secret-name: ALPACA_SECRET_KEY
+        secret-value: "${{ env.ALPACA_SECRET_KEY }}"
```
[the remark this answers is not in the document]
Comment (saikrishna-dev) [reply .github/actions/setup-secrets/action.yml:83]: Skipping this one intentionally — the empty-but-present secret is benign and gating only Alpaca would break the established pattern:

- Every optional secret in `setup-secrets/action.yml` (`SERP_API_KEY`, `X_AI_API_KEY`) is already created unconditionally with the same `secret-value: "${{ env.X }}"` form. `SERP_API_KEY` is documented optional and has shipped this way without breaking the Koyeb deploy, so empty Alpaca secrets behave identically.
- An empty value resolves to an empty env var, and the backend handles that gracefully: `market.py` `if not Config.ALPACA_API_KEY or not Config.ALPACA_SECRET_KEY:` returns the documented 503 "not configured" for stocks. Crypto (Coinbase) needs no key and is unaffected.
- Truly gating it would also require dynamically omitting the `@ALPACA_*` entries from the static `service-env` string — a disproportionate refactor for no functional gain.

Net: keeping it consistent with the other optional secrets. Will revisit if we ever make Alpaca mandatory.

---
_Generated by [Claude Code](https://claude.ai/code)_
